tests/Report/AbstractReporterTestCase.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Report;
+
+use PhpCsFixer\Report\ReporterInterface;
+use PhpCsFixer\Report\ReportSummary;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * @internal
+ */
+abstract class AbstractReporterTestCase extends TestCase
+{
+    /**
+     * @var ReporterInterface
+     */
+    protected $reporter;
+
+    protected function setUp()
+    {
+        $this->reporter = $this->createReporter();
```

Comment:
what for? we don't do it in any of tests

Reply:
To reduce memory usage. Even if here it should not be a big amount of memory used, I think it's a good practice to avoid using properties in tests as much as possible, and clean them otherwise.

Reply:
just tried it with
```php
    protected function tearDown()
    {
        unset($this->reporter);
    }
```

no change in memory usage

also, if creating data not unset in `tearDown` affects memory, we shall clean it up globally (not by creating a lot of `setUp` concrete methods manually).
Let us discuss it separatelly

Reply:
Ok then. Maybe using properties this way doesn't always affect memory usage much.